templates/nextflow-s3-template.yaml

```diff
@@ -0,0 +1,148 @@
+AWSTemplateFormatVersion: 2010-09-09
+Description: Private S3 Bucket
+
+# Parameters
+Parameters:
+  Department:
+    Description: 'The department for this resource'
+    Type: String
+    AllowedPattern: '^\S*$'
+    ConstraintDescription: 'Must be string with no spaces'
+  Project:
+    Description: 'The name of the project that this resource is used for'
+    Type: String
+    AllowedPattern: '^\S*$'
+    ConstraintDescription: 'Must be string with no spaces'
+  OwnerEmail:
+    Description: 'Email address of the owner of this resource'
+    Type: String
+    AllowedPattern: '^(([^<>()\[\]\\.,;:\s@"]+(\.[^<>()\[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$'
+    ConstraintDescription: 'Must be an acceptable email address syntax(i.e. joe.smith@sagebase.org)'
+  BucketName:
+    Type: String
+    Description: (Optional) Name of the created bucket.
+  ReadWriteAccess:
+    Type: CommaDelimitedList
+    Default: "[]"
+    Description: (Optional) Grant read/write access to accounts, groups, and users.
+    ConstraintDescription: List of ARNs (i.e. ["arn:aws:iam::011223344556:user/jsmith", "arn:aws:iam::544332211006:user/rjones"]
+  ReadOnlyAccess:
+    Type: CommaDelimitedList
+    Default: "[]"
+    Description: (Optional) Grant read access to accounts, groups, and users.
+    ConstraintDescription: List of ARNs (i.e. ["arn:aws:iam::011223344556:user/jsmith", "arn:aws:iam::544332211006:user/rjones"]
+
+# Conditions
+Conditions:
+  HasBucketName: !Not [!Equals [!Ref BucketName, ""]]
+
+# Mappings
+Mappings:
+  Accounts:
+    "035458030717":
+      AdminRole: "arn:aws:iam::035458030717:role/aws-reserved/sso.amazonaws.com/AWSReservedSSO_Administrator_580e9f32ac55c4e7"
+    "728882028485":
+      AdminRole: "arn:aws:iam::728882028485:role/aws-reserved/sso.amazonaws.com/AWSReservedSSO_Administrator_3a9d61d0884be260"
+
+# Resources
+Resources:
+  Bucket:
+    Type: AWS::S3::Bucket
+    Properties:
+      BucketName: !If [HasBucketName, !Ref BucketName, !Ref 'AWS::NoValue']
```

Comment:
Out of curiosity, can this be achieved by setting the default value of the `BucketName` parameter to `AWS::NoValue`? I'm not requiring that change here, but I'm curious so we can simplify templates going forward.

Reply:
No, because you'd have to use an intrinsic function for this, and they're not allowed in Parameters.

Reply:
Thanks for clarifying, @tthyer!